sql-plugin/src/main/311until320-all/scala/com/nvidia/spark/rapids/shims/ShuffleOriginUtil.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import scala.collection.immutable.HashSet
+
+import org.apache.spark.sql.execution.exchange.{ENSURE_REQUIREMENTS, REPARTITION, REPARTITION_WITH_NUM, ShuffleOrigin}
+
+object ShuffleOriginUtil {
+  def getSupportedShuffleOrigins: HashSet[ShuffleOrigin] = { 
+    HashSet(ENSURE_REQUIREMENTS, REPARTITION, REPARTITION_WITH_NUM)
+  }
```

Comment:
This still constructs the hash set on every invocation.  We only need to construct it once for the life of the process.  I recommend an approach like this:
```scala
  private val knownOrigins = HashSet(ENSURE_REQUIREMENTS, ...)

  def isSupported(origin: ShuffleOrigin): Boolean = knownOrigins.contains(origin)
```